CorsixTH/Lua/objects/doors/swing_door_right.lua

```diff
@@ -38,22 +38,35 @@ local SwingDoor = _G["SwingDoor"]
 function SwingDoor:SwingDoor(hospital, object_type, x, y, direction, etc)
   self.is_master = object_type == object
   self:Door(hospital, object_type, x, y, direction, etc)
-  if self.is_master then
-    local slave_type = "swing_door_left"
-    self.slave = self.world:getObject(x - 1, y, slave_type) or self.world:getObject(x, y - 1, slave_type) or nil
-    self.slave:setAsSlave(self)
-    assert(self.slave.master, "Swing doors are not paired at " .. x .. "," .. y)
-  end
+  self:pairDoors(x, y)
   self.old_anim = self.th:getAnimation()
   self.old_flags = self.th:getFlag()
   local --[[persistable:swing_door_creation]] function _() end -- Stub from Oct 2023, savegame version 181
 end
 
---[[ Makes the door mimic its master when it comes to hover cursor and what happens
-when the player clicks on it.
+--! Links the master to the slave swing door. This allows the slave to mimic the
+--! master in interactions
+--!param x (num) Co-ordinate
+--!param y (num) Co-ordinate
+function SwingDoor:pairDoors(x, y)
+  -- Because we don't know the build order of each door we must accomodate for this
+  if self.is_master then -- The master will check for an adjacent slave
+   local slave_type = "swing_door_left"
+   self.slave = self.world:getObject(x - 1, y, slave_type) or self.world:getObject(x, y - 1, slave_type) or nil
+
+   if self.slave then
+     self.slave.master = self
+   end
+ else -- The slave will check for an adjacent master
+   local master_type = "swing_door_right"
+   self.master = self.world:getObject(x + 1, y, master_type) or self.world:getObject(x, y + 1, master_type) or nil
+
+   if self.master then
+     self.master.slave = self
+   end
+ end
```

Comment:
```suggestion
    local slave_type = "swing_door_left"
    self.slave = self.world:getObject(x - 1, y, slave_type) or self.world:getObject(x, y - 1, slave_type)

    if self.slave then
      self.slave.master = self
    end
  else -- The slave will check for an adjacent master
    local master_type = "swing_door_right"
    self.master = self.world:getObject(x + 1, y, master_type) or self.world:getObject(x, y + 1, master_type)

    if self.master then
      self.master.slave = self
    end
  end
  assert((self.master and self.master.slave) or (self.slave and self.slave.master),
      "Swing doors did not pair at " .. x .. "," .. y)
```
Added the assert line, took out "or nil", and a minor whitespace fix. This works on the saves I had from before - wards and operating theatres with doors facing all directions

Reply:
What happens on our first run of this? Generally speaking only one half of the door exists so it shouldn't be able to set `self.master.slave` nor `self.slave.master`.

On testing, the assert does crash on building a new room:
```
An error has occurred!
Almost anything can be the cause, but the detailed information below can help the developers find the source of the error.
Running: The buttonup handler.
A stack trace is included below, and the handler has been disconnected.
Lua\objects/doors\swing_door_right.lua:66: Swing doors did not pair at 60,63
stack traceback:
        [C]: in function 'assert'
        Lua\objects/doors\swing_door_right.lua:66: in method 'pairDoors'
```

Reply:
Yes, I didn't test well.